web/ui/react-app/src/pages/rules/RulePanel.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC, CSSProperties, Fragment } from 'react';
+import { Rule } from '../../types/types';
+import { Link } from '@reach/router';
+import { createExpressionLink, isPresent } from '../../utils';
+
+interface RulePanelProps {
+  rule: Rule;
+  styles?: CSSProperties;
+  tag?: keyof JSX.IntrinsicElements;
+}
+
+export const RulePanel: FC<RulePanelProps> = ({ rule, styles = {}, tag }) => {
+  const Tag = tag || Fragment;
+  const { name, query, labels, annotations, type, duration } = rule;
+  const style = { background: '#f5f5f5', ...styles };
+  return (
+    <Tag {...(tag ? { style } : {})}>
+      <pre className="m-0" style={tag ? undefined : style}>
+        <code>
+          <div>
+            {/* Type of the rule is either alerting or recording. */}
+            {type === 'alerting' ? 'alert: ' : 'record: '}
+            <Link to={createExpressionLink(`ALERTS{alertname="${name.replace(/([/"])/g, "\\$1")}"}`)}>{name}</Link>
```

Comment:
Wouldn't you just need:
```suggestion
            <Link to={createExpressionLink(`ALERTS{alertname="${name.replace(/"/g, '\\"')}"}`)}>{name}</Link>
```

Reply:
yes, thanks - it's better